### PR TITLE
fix worker error in macos #5513

### PR DESCRIPTION
**Pull request by gaecoli**
## What type of PR is this? 

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## How is this tested?

- [ ] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [ ] Manually
- [ ] N/A

### CODE REVIEW

**Comment by gaecoli** [comment]
image


In macos.

**Review by junnplus — APPROVED** [review]
LGTM

**Comment by konnectr** [comment]
It is probably advisable to use text so that people can find the problem if they encounter it again later

**Comment by gaecoli** [comment]
> It is probably advisable to use text so that people can find the problem if they encounter it again later

What's meaning?

**Comment by gaecoli** [comment]
#5513

**Comment by gaecoli** [comment]
> It is probably advisable to use text so that people can find the problem if they encounter it again later


I may not understand what you mean, what type of text are you talking about?

**Comment by konnectr** [comment]
text instead of pictures

**Comment by gaecoli** [comment]
> text instead of pictures

Ok! You are right!

**Comment by justinclift** [comment]
> text instead of pictures

Text is also useful for search results. :smile:

**Comment by justinclift** [comment]
Uh oh.  We need to be careful when doing PR's like this.  It looks like a lot of the PR was first created by @RajatGoyal.

However, none of the commits in this PR belong to @RajatGoyal any more.  This can look like us stealing other peoples work, which is very very bad for an Open Source project.

**Comment by justinclift** [comment]
I'm going to revert this commit, so we can instead update the other PR with the changes here.

**Comment by gaecoli** [comment]
> I'm going to revert this commit, so we can instead update the other PR with the changes here.

Yes! I agree with you!

**Comment by gaecoli** [comment]
Connected to pydev debugger (build 222.3739.56)
[2023-07-21 17:01:53,154][PID:19782][INFO][xmlschema] Resource 'XMLSchema.xsd' is already loaded
[2023-07-21 17:01:54,687][PID:19782][INFO][rq.worker] Worker rq:worker:16252905b1824cb69c5eeeeee143cddc: started, version 1.5.0
[2023-07-21 17:01:54,688][PID:19782][INFO][rq.worker] *** Listening on periodic, emails, default, scheduled_queries, queries, schemas...
[2023-07-21 17:01:58,633][PID:19782][INFO][rq.worker] default: 32474c9c-1f46-4c17-a23d-d6fee7179547
[2023-07-21 17:01:58,764][PID:20349][ERROR][rq.worker] Worker rq:worker:16252905b1824cb69c5eeeeee143cddc: found an unhandled exception, quitting...
Traceback (most recent call last):
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 552, in work
    self.execute_job(job, queue)
  File "/Users/guyu/redash-11/redash/redash/tasks/worker.py", line 64, in execute_job
    super().execute_job(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 775, in execute_job
    self.fork_work_horse(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 705, in fork_work_horse
    self.main_work_horse(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 785, in main_work_horse
    self.setup_work_horse_signals()
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 1079, in setup_work_horse_signals
    signal.signal(signal.SIGRTMIN, self.request_stop_sigrtmin)
AttributeError: module 'signal' has no attribute 'SIGRTMIN'
[2023-07-21 17:01:59,692][PID:19782][INFO][rq.worker] default: d54f3d5d-4a4a-4ed6-ac5d-64f7ed8f81ad
[2023-07-21 17:01:59,818][PID:20351][ERROR][rq.worker] Worker rq:worker:16252905b1824cb69c5eeeeee143cddc: found an unhandled exception, quitting...
Traceback (most recent call last):
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 552, in work
    self.execute_job(job, queue)
  File "/Users/guyu/redash-11/redash/redash/tasks/worker.py", line 64, in execute_job
    super().execute_job(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 775, in execute_job
    self.fork_work_horse(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 705, in fork_work_horse
    self.main_work_horse(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 785, in main_work_horse
    self.setup_work_horse_signals()
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 1079, in setup_work_horse_signals
    signal.signal(signal.SIGRTMIN, self.request_stop_sigrtmin)
AttributeError: module 'signal' has no attribute 'SIGRTMIN'
[2023-07-21 17:02:00,751][PID:19782][INFO][rq.worker] queries: 46c8346c-7a2a-4121-80cc-43b8d3fd556a
[2023-07-21 17:02:00,874][PID:20352][ERROR][rq.worker] Worker rq:worker:16252905b1824cb69c5eeeeee143cddc: found an unhandled exception, quitting...
Traceback (most recent call last):
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 552, in work
    self.execute_job(job, queue)
  File "/Users/guyu/redash-11/redash/redash/tasks/worker.py", line 64, in execute_job
    super().execute_job(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 775, in execute_job
    self.fork_work_horse(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 705, in fork_work_horse
    self.main_work_horse(job, queue)
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 785, in main_work_horse
    self.setup_work_horse_signals()
  File "/Users/guyu/redash-11/redash/venv/lib/python3.9/site-packages/rq/worker.py", line 1079, in setup_work_horse_signals
    signal.signal(signal.SIGRTMIN, self.request_stop_sigrtmin)
AttributeError: module 'signal' has no attribute 'SIGRTMIN'

**Comment by konnectr** [comment]
> I'm going to revert this commit, so we can instead update the other PR with the changes here.

You're right.

**Comment by gaecoli** [comment]
@justinclift Please help me revert it. I'm sorry to do that! Thank you! This code i add in my local development env, so i'm not focus @RajatGoyal PR. Sorry about that!

**Comment by justinclift** [comment]
It's ok, we can fix it. :grin:

I've just manually force pushed to our master branch, without this PR on the end of it.  So, it's like it never got merged. :wink:

@gaecoli Please reopen that existing PR, and add your changes to it as new commits.  That way, when we merge it @RajatGoyal will be correctly in our commit history. :smile:

**Comment by gaecoli** [comment]
> It's ok, we can fix it. 😁
> 
> I've just manually force pushed to our master branch, without this PR on the end of it. So, it's like it never got merged. 😉
> 
> @gaecoli Please reopen that existing PR, and add your changes to it as new commits. That way, when we merge it @RajatGoyal will be correctly in our commit history. 😄

I edit @RajatGoyal PR? How to add in his PR?

**Comment by justinclift** [comment]
Yep.  Because you have `maintainer` status in our GitHub repository, it means you can push to the branches for other people's (Redash) PRs.

There are a few ways to do it.

This is one example, from the command line:

## 1. Add a "remote" repository

```
$ git remote add rajat https://github.com/RajatGoyal/redash
$ git remote update rajat
```

## 2. Checkout the `rajat/fix/macos-rq-worker` branch, then give it a name

```
$ git checkout rajat/fix/macos-rq-worker
$ git switch -c temp1
```

You are now in a new `temp1` branch, that has the commits from the PR in it.

## 3. Make changes

You can make changes now, and add your commits here.

```
$ <edit files>
$ git commit
```

## 4. Push to remote repository

```
$ git push rajat temp1:fix/macos-rq-worker
```

**Comment by justinclift** [comment]
@gaecoli If you hit any problems then let me know.  I can probably get it done if we need. :smile:

**Comment by gaecoli** [comment]
> @gaecoli If you hit any problems then let me know. I can probably get it done if we need. 😄

remote reject me.

**Comment by justinclift** [comment]
Ahhh ok.  Lets do it in a new, temporary branch in our repo.  When we have the commits all ok, then we can figure out the PR part.

Give me a few minutes as I need to get food first. :smile:

**Comment by justinclift** [comment]
Use this branch instead: https://github.com/getredash/redash/tree/macos_fix_pr_6222

When you are finished with it, let me know and I'll get this PR updated to use it. :smile:

**Comment by gaecoli** [comment]
> Ahhh ok. Lets do it in a new, temporary branch in our repo. When we have the commits all ok, then we can figure out the PR part.
> 
> Give me a few minutes as I need to get food first. 😄



> Use this branch instead: https://github.com/getredash/redash/tree/macos_fix_pr_6222
> 
> When you are finished with it, let me know and I'll get this PR updated to use it. 😄

You can use my code. I'm going to be busy for a while.

**Comment by justinclift** [comment]
Ok, where is your code?

File: redash/tasks/worker.py [hunk 1]
```diff
@@ -10,9 +11,15 @@
 from rq.worker import (
     HerokuWorker,  # HerokuWorker implements graceful shutdown on SIGTERM
 )
+from rq.worker import Worker
```
Comment (gaecoli) [inline redash/tasks/worker.py:14]: @justinclift There.

Comment (justinclift) [reply redash/tasks/worker.py:14]: Thanks. :smile:

**Pull request merged**